tst/EnergyPlus/unit/WaterThermalTanks.unit.cc

```diff
@@ -2471,3 +2471,120 @@ TEST_F(EnergyPlusFixture, DesuperheaterTimeAdvanceCheck){
     EXPECT_GT(Tank.TankTemp, 50);
 
 }
+
```

Comment:
I was expecting this unit test to include a direct call to `SourceHeatNeed` since you extracted it out into a worker function, but I guess it's called from within other calls below.

Reply:
Yes it's calling ``PlantMassFlowRatesFun`` inside ``InitWaterThermalTank`` which is not straghtforward. It's a good idea to directly hit this function. I addressed this in the new commits below.